apps/frontend/src/pages/oauth/exchange_token.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {useEffect} from "react";
+import {useRouter} from "next/router";
+import {mintOAuthToken} from "@/lib/oauth";
+import {toast} from "sonner";
+import {CursiveLogo} from "@/components/ui/HeaderCover";
+import {AccessToken} from "@types";
+import {saveAccessToken} from "@/lib/storage/localStorage/oauth";
+
+const OAuthAccessTokenPage: React.FC = () => {
+  const router = useRouter();
+
+  useEffect(() => {
+    const fetchAccessToken = async () => {
+      const { code, state } = router.query;
+
+      // TODO: need to exit if code / state missing
```

Comment:
How can I handle a case where the code or state query parameters are missing? I'm trying to use setTimeout / clearTimeout but I'm running into node:timer compilation failures.